### PR TITLE
Lifecycle rules for s3 and gcp

### PR DESCRIPTION
**Pull request by akira**

### CODE REVIEW

File: versioned_docs/version-2.0/self_hosting/configuration/blob_storage.mdx [hunk 1]
```diff
@@ -111,3 +112,74 @@ If using an access key and secret, you can also provide an existing Kubernetes s
 This is recommended over providing the access key and secret key directly in your config.
 
 :::
+
+## TTL Configuration
+
+If using the [TTL](/self_hosting/configuration/ttl) feature with LangSmith, you'll also have to configure TTL rules for
+your blob storage. Trace information stored on blob storage is stored on a particular prefix path, which determines the TTL for the
+data. As traces are upgrades, they are place in the upgrade prefix path as well to ensure that the data is upgraded in
```
Comment (jakerachleff) [inline versioned_docs/version-2.0/self_hosting/configuration/blob_storage.mdx:120]: I'm confused what you mean here

Comment (jakerachleff) [reply versioned_docs/version-2.0/self_hosting/configuration/blob_storage.mdx:120]: Do you mean something like:

```
When a trace's retention is extended, its corresponding blob storage path changes to ensure that it has a path that matches the new extended retention. 
```

File: versioned_docs/version-2.0/self_hosting/configuration/blob_storage.mdx [hunk 2]
```diff
@@ -111,3 +112,74 @@ If using an access key and secret, you can also provide an existing Kubernetes s
...
+blob storage as well.
+
+The following TTL prefix are used:
+
+- `ttl_s/`: Short term TTL, configured for 15 days.
+- `ttl_l/`: Long term TTL, configured for 401 days.
```
Comment (jakerachleff) [inline versioned_docs/version-2.0/self_hosting/configuration/blob_storage.mdx:126]: Worth noting 1 day longer than TTLs in app?

Comment (jakerachleff) [reply versioned_docs/version-2.0/self_hosting/configuration/blob_storage.mdx:126]: and if that you customize them, then you should also follow the same pattern?

Comment (langchain-infra) [reply versioned_docs/version-2.0/self_hosting/configuration/blob_storage.mdx:126]: ^ agreed with adding note on customizing

Comment (akira) [reply versioned_docs/version-2.0/self_hosting/configuration/blob_storage.mdx]: We actually cutoff the TTL display from clickhouse via query, so I will actually set it to be same as clickhouse.  Added note on customization.

File: versioned_docs/version-2.0/self_hosting/configuration/blob_storage.mdx (hunk 1 above)
Comment (akira) [reply versioned_docs/version-2.0/self_hosting/configuration/blob_storage.mdx]: Update: 

`When a trace's retention is extended, its corresponding blob storage path changes to ensure that it matches the new extended retention.`

**Review by jakerachleff — APPROVED** [review]

**Pull request merged**